src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2262,10 +2541,99 @@ bool TheoryStringsRewriter::checkEntailArith(Node a, bool strict)
       return true;
     }
     // TODO (#1180) : abstract interpretation goes here
+
+    // over approximation O/U
+
+    // O( x <op> y ) -> O( x ) <op> O( y )
+    // O( len( x ) ) -> ...
+    // O( len( int.to.str( x ) ) ) -> ...
+    // O( len( str.substr( x, n1, n2 ) ) ) -> O( n2 ) | O( len( x ) )
+    // O( len( str.replace( x, y, z ) ) ) -> 
+    //   O( len( x ) ) + O( len( z ) ) - U( len( y ) )
+    // O( indexof( x, y, n ) ) -> O( len( x ) ) - U( len( y ) )
+    // O( str.to.int( x ) ) -> ...
+
+    // U( x <op> y ) -> U( x ) <op> U( y )
+    // U( len( x ) ) -> ...
+    // U( len( int.to.str( x ) ) ) -> 1
+    // U( len( str.substr( x, n1, n2 ) ) ) -> 
+    //   min( U( len( x ) ) - O( n1 ), U( n2 ) )
+    // U( len( str.replace( x, y, z ) ) ) -> 
+    //   U( len( x ) ) + U( len( z ) ) - O( len( y ) ) | 0
+    // U( indexof( x, y, n ) ) -> -1    ?
+    // U( str.to.int( x ) ) -> -1
```

Comment:
Can `str.to.int` not convert negative numbers? In that case this should be 0 otherwise it should be -infinity.

Reply:
It doesn't convert negative numbers currently, so -1 is indeed a lower bound.  This could change when the smt2 standard for strings is developed though.

Reply:
But 0 would be a tighter bound, no?

Reply:
We currently are using a total semantics where str.to.int( x ) = -1 if x is not a positive number.  So, currently the bound is -1, although this could change if the semantics change later.